packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -57,28 +58,39 @@ import {resolveSiteFromUrl} from '../../utils/site-utils'
 import useMultiSite from '../../hooks/use-multi-site'
 import {useCategory, useCustomerType} from 'commerce-sdk-react-preview'
 
-const App = (props) => {
-    const {children, targetLocale = DEFAULT_LOCALE, messages = {}} = props
-    const {data: allCategories} = useCategory(
-        {parameters: {id: CAT_MENU_DEFAULT_ROOT_CATEGORY, levels: CAT_MENU_DEFAULT_NAV_DEPTH}},
+const onClient = typeof window !== 'undefined'
+
+/* 
+The categories tree can be really large! For performance reasons,
+we only load the level 0 categories on server side, and load the rest
+on client side to reduce SSR page size.
+*/
+const useLazyLoadCategories = () => {
+    const allCategoriesQuery = useCategory(
+        {parameters: {id: CAT_MENU_DEFAULT_ROOT_CATEGORY, levels: CAT_MENU_DEFAULT_NAV_CLIENT_DEPTH}},
         {
-            select: (categories) => {
-                // Note: What is the best to handle special case like this?? Should commerce sdk handles this?
-                if (categories.isError) {
-                    const message =
-                        categories.title === 'Unsupported Locale'
-                            ? `
-It looks like the locale “${categories.locale}” isn’t set up, yet. The locale settings in your package.json must match what is enabled in your Business Manager instance.
-Learn more with our localization guide. https://sfdc.co/localization-guide
-`
-                            : categories.detail
-                    throw new Error(message)
-                }
-                return flatten(categories, 'categories')
-            }
+            enabled: onClient
         }
     )
 
+    const levelZeroCategoriesQuery = useCategory({
+        parameters: {id: CAT_MENU_DEFAULT_ROOT_CATEGORY, levels: CAT_MENU_DEFAULT_NAV_SSR_DEPTH},
+    }, {
+        enabled: !allCategoriesQuery.isFetched
```

Comment:
`isFetched` is only true once the query has completed, meaning `levelZero` will trigger while `allCategories` is still loading. If the page is being served with `levelZero` already loaded from SSR, then it seems unnecessary to have it load again, just to be replaced.

If it's not desired behavior, you could do just a single query:
```js
useCategory({
  parameters: {
    id: CAT_MENU_DEFAULT_ROOT_CATEGORY,
    levels: onClient ? CAT_MENU_DEFAULT_NAV_CLIENT_DEPTH : CAT_MENU_DEFAULT_NAV_SSR_DEPTH
  }
})
```